model/doc.go

```diff
@@ -0,0 +1,2 @@
+// Package model defines Evergreen's data model.
```

Comment:
I think this one actually deserves more explanation since it's fairly complicated and has lots of random, unrelated things in it since it's _all_ the Evergreen DB models. I think this package comment is not that useful unless it actually enumerates some of the things here. For example:

* Can we call out what the subpackages represent?
* There's a lot of unrelated random things in the models (top-level) package. There's stuff related to task lifecycle,   project YAML config parsing (e.g. the parser project), github integration, etc.